client/src/App.js

```diff
@@ -14,15 +15,25 @@ export default class App extends Component {
 
   addToSavedList = movie => {
     const savedList = this.state.savedList;
-    savedList.push(movie);
-    this.setState({ savedList });
+
+    if (!JSON.stringify(savedList).includes(`"id":${movie.id}`)) {
```

Comment:
This is an interesting way of doing this! Typically you'd expect something more along the lines of just sticking to an array or searching for keys on _one level only_. While this works fine for this structure, this sort of thing could run into trouble if you had something nested with an id key that just happened to have the same id as a movie (while the id isn't actually on a movie).